cmd/bad-key-revoker/main.go

```diff
@@ -121,18 +121,18 @@ func (bkr *badKeyRevoker) findUnrevoked(unchecked uncheckedBlockedKey) ([]unrevo
 				&unrevokedCert,
 				`SELECT cs.id, cs.serial, c.registrationID, c.der
 				FROM certificateStatus AS cs
-				JOIN certificates AS c
+				LEFT JOIN precertificates AS c
 				ON cs.serial = c.serial
 				WHERE cs.serial = ? AND cs.isExpired = false AND cs.status != ?`,
```

Comment:
Do you not now have to do both the is expired, and status != revoked checks in code? What we get from keyHashToSerial has no guarantee to satisfy these two checks, which will cause errors below because you've removed the IsNoRows check

Reply:
Seems like we actually want an INNER JOIN here, which would solve this problem, and mean we don't need to check for the DER column being NULL.